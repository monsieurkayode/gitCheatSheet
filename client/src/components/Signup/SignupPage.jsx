import React, { Fragment } from 'react';
import Container from '../Common/Container';
import Button from '../Common/Button';
import FormField from '../Common/FormField';

const LoginPage = () => (
  <Fragment>
    <Container>
      <form>
        <FormField
          icon="account_circle"
          placeholder="Username"
          name="username"
          value=""
          error="Username is required"
        />
        <FormField
          icon="lock"
          type="password"
          placeholder="Password"
          name="password"
          value=""
          error="Password is required"
        />
        <FormField
          icon="lock"
          type="password"
          placeholder="Confirm Password"
          name="password"
          value=""
          error="Passwords do not match"
        />
        <div className="form-field">
          <Button type="button" value="Register" />
        </div>
      </form>
    </Container>
  </Fragment>
);

export default LoginPage;
