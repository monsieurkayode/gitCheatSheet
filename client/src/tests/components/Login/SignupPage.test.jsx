import React from 'react';
import { shallow } from 'enzyme';
import { SignupPage, mapStateToProps } from '../../../components/Signup/SignupPage';

const setup = () => {
  const props = {
    authenticated: false,
    loading: false,
    registerUser: () => Promise.resolve()
  };

  return shallow(<SignupPage {...props} />);
};

describe('SignupPage Component', () => {
  const wrapper = setup();
  const event = { target: { name: 'username', value: '' }, preventDefault: jest.fn() };

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleInputChange', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleInputChange');
    wrapper.instance().handleInputChange(event);
    expect(spy).toHaveBeenCalled();
  });

  describe('handleSubmit', () => {
    it('should be called', () => {
      const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');
      wrapper.instance().handleSubmit(event);
      expect(spy).toHaveBeenCalled();
    });

    it('should set error if form is invalid', () => {
      wrapper.instance().handleSubmit(event);
      expect(wrapper.state('errors'))
        .toEqual({
          password: 'Password is required',
          username: 'Username is required'
        });
    });

    it('should submit valid forms with no errors', () => {
      wrapper.setState({
        user: { username: 'user', password: 'password', confirmPassword: 'password' },
        errors: {}
      });
      wrapper.instance().handleSubmit(event);
      expect(wrapper.state('errors')).toEqual({});
    });
  });


  it('should call handleBlur', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleBlur');
    wrapper.instance().handleBlur(event);
    expect(spy).toHaveBeenCalled();
  });

  it('should return a Redirect component if user is authenticated', () => {
    wrapper.setProps({ authenticated: true });
    expect(wrapper.find('Redirect').length).toBe(1);
  });

  describe('mapStateToProps', () => {
    it('should return an object', () => {
      expect(mapStateToProps({ userDetails: { user: {}, makingApiRequest: true } }))
        .toBeInstanceOf(Object);
    });
  });
});
