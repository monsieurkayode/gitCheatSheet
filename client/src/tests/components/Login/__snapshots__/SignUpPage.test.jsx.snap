// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage Component should render correctly 1`] = `
<Fragment>
  <styled.main>
    <form
      onSubmit={[Function]}
    >
      <FormField
        error=""
        icon="account_circle"
        loading={false}
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <FormField
        error=""
        icon="lock"
        loading={false}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <FormField
        error=""
        icon="lock"
        loading={false}
        name="confirmPassword"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Confirm Password"
        type="password"
        value=""
      />
      <div
        className="form-field"
      >
        <styled.input
          disabled={false}
          type="submit"
          value="Register"
        />
        <Loader
          inline={true}
          loading={false}
          size={15}
          strokeColor="#fff"
        />
      </div>
    </form>
  </styled.main>
</Fragment>
`;
